tapir/coop/views/shareowner.py

```diff
@@ -567,3 +573,84 @@ def get_queryset(self):
             .exclude(willing_to_gift_a_share=None)
             .order_by("willing_to_gift_a_share")
         )
+
+
+class ShareOwnerTableWelcomeDesk(django_tables2.Table):
+    class Meta:
+        model = ShareOwner
+        template_name = "django_tables2/bootstrap4.html"
+        fields = [
+            "id",
+        ]
+        sequence = (
+            "id",
+            "display_name",
+        )
+        order_by = "id"
+
+    display_name = django_tables2.Column(
+        empty_values=(), verbose_name="Name", orderable=False
+    )
+
+    def render_display_name(self, value, record: ShareOwner):
+        return format_html(
+            "<a href={}>{}</a>",
+            reverse("coop:welcome_desk_member", args=[record.pk]),
+            record.get_info().get_display_name(),
+        )
+
+
+class ShareOwnerFilterWelcomeDesk(django_filters.FilterSet):
+    display_name = CharFilter(
+        method="display_name_filter", label=_("Name or member ID")
+    )
+
+    def display_name_filter(
+        self, queryset: ShareOwner.ShareOwnerQuerySet, name, value: str
+    ):
+        if not value:
+            return queryset.none()
+
+        if value.isdigit():
+            return queryset.filter(id=int(value))
+
+        return queryset.with_name(value)
+
+
+class WelcomeDeskSearchView(PermissionRequiredMixin, FilterView, SingleTableView):
+    permission_required = "accounts.view"
+    template_name = "coop/welcome_desk_search.html"
+    table_class = ShareOwnerTableWelcomeDesk
+    model = ShareOwner
+    filterset_class = ShareOwnerFilterWelcomeDesk
+
+
+class WelcomeDeskMemberView(PermissionRequiredMixin, generic.DetailView):
```

Comment:
(nit) With "member" we are introducing yet another word. Why not stick with ShareOwner?